### ISSUE TITLE
Column default value as function

### ISSUE THREAD

**Issue by CocaColaBear** [typeorm/typeorm#150]
I have column Id:
```
@PrimaryColumn({
    name: "id",
    type: "bigint",
    default: new_id() // new_id() is function in DB (postgresql)
  })
  Id: number;
```
In database i have function: new_id() for generate ids. How i can implement this query?
`id bigint NOT NULL DEFAULT new_id()`

**Comment by pleerock** [comment]
```typescript
@PrimaryColumn({
    name: "id",
    type: "bigint",
    default: "new_id()"
  })
  Id: number;
```

Doesnt it work?

**Comment by CocaColaBear** [comment]
@pleerock  not work
Error:
```
{ error: invalid input syntax for integer: "new_id()"
    at Connection.parseE (C:\TS\Test\node_modules\pg\lib\connection.js:554:11)
    at Connection.parseMessage (C:\TS\Test\node_modules\pg\lib\connection.js:381:17)
    at Socket.<anonymous> (C:\TS\Test\node_modules\pg\lib\connection.js:117:22)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:551:20)
  name: 'error',
  length: 94,
  severity: 'ERROR',
  code: '22P02',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'int8.c',
  line: '99',
  routine: 'scanint8' }
```

**Comment by pleerock** [comment]
mmm probably because of escaping... Probably we can add another option like `preventDefaultEscaping: boolean`, but I would like to hear better alternative @eduardoweiland @19majkel94 do you have any insights about this issue?

**Comment by CocaColaBear** [comment]
@pleerock mb it is dirty, but we can check default value : function or not (regexp or another way).
But option 'preventDefaultEscaping' looks nice. I can make pull request with this option

**Comment by pleerock** [comment]
> function or not (regexp or another way)

can we cover all cases with such regexp? 

I would like to hear something better then `preventDefaultEscaping`, maybe other guys have some opinions before you will make a PR with dirty `preventDefaultEscaping` option.

**Comment by eduardoweiland** [comment]
I have an idea, but maybe it's too much complex. Wrap the values that are sent to the database in some class, like `EscapedValue` and `RawSql`. These classes would hold the logic to convert the value to be sent to the database. I don't know how this would work with multiple drivers, but it's a starting point. What do you think?

**Comment by 19majkel94** [comment]
5000 option and parameters makes the decorators code big and gross. I would preffer simple solutions like `&`, `@`, `=`, `<` in `AngularJS` or in template engines like `pug`.

I don't know how about TypeORM v0.0.6 but the older version required quotes in string for default values:
```typescript
@Column("string", { default: "'normal'" })
```
so we could also make this part of code more inteligent - if field is string, wrap argument in quotes in sql, if number no, etc.

My proposition are twos:
1. Overloading decorator - you can pass function returning an unescaped string:
```typescript
@PrimaryColumn({ default: () => "new_id()" })
```
it could be even `default: unescaped => "new_id()"` or sth like this.

2. Parse string - if first char is '=' (or '!'), don't escape string. But it's like old JS style.

It could also be an alternative to `preventDefaultEscaping` option :wink:

**Comment by pleerock** [comment]
Thank you guys for the feedback.

`preventDefaultEscaping` really looks shitty beside your ideas.

@eduardoweiland This is really cool idea, much better then `preventDefaultEscaping`.

@19majkel94 I would like to prevent extra rules like "!" or "&" which aren't intuitive. But you bring a valueable point, what if value is lets say `TRUE` or `FALSE` or number, or string, or function.

I think better if we won't do escaping at all, and those who want to use strings should put single quotes. It will solve all the problems and its intuitive enought (but still we need point this in docs).

**Comment by pleerock** [comment]
Ive made a decision to use @19majkel94 suggested solution to use `function notation` for default values for now. It will work this way:

```typescript
@Column({ default: "Umed" })
name: string;
```
generates `DEFAULT 'Umed'`

```typescript
@Column({ default: 1 })
count: number;
```
generates `DEFAULT 1`

```typescript
@Column({ default: true })
isSomething: boolean;
```
generates `DEFAULT TRUE`

```typescript
@Column({ default: () => "pow(5)" })
count: number;
```
generates `DEFAULT pow(5)`

This fix released in `0.0.7-alpha.11`
